### PR TITLE
Ignore nightly tags in release notes

### PR DESCRIPTION
**Pull request by samdark**
## Summary
- make release notes find the previous stable version tag only
- render changelog authors as @username-style handles instead of real names
- add packaging regression tests so nightly tags are not used as changelog boundaries and names do not come from %an

## Cause
After immutable nightly releases, `git describe --tags --abbrev=0 "${tag}^"` can return a `nightly-*` tag. That makes release notes show only changes since the latest nightly instead of the previous stable release. Existing notes also used real author names from `%an`; they should use handles.

## Existing releases
Updated the existing stable GitHub release descriptions for `0.1.0` and `0.1.1` to use the new format. Nightly prerelease descriptions were left unchanged.

## Tests
- `make test tests/Unit/Packaging/ConfigurationPackagingTest.php`
- `make psalm`
- `make composer-dependency-analyser`

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -266,7 +266,7 @@ jobs:
         run: |
           set -euo pipefail
           tag="${GITHUB_REF_NAME}"
-          previous_tag="$(git describe --tags --abbrev=0 "${tag}^" 2>/dev/null || true)"
+          previous_tag="$(git describe --tags --abbrev=0 --match '[0-9]*.[0-9]*.[0-9]*' "${tag}^" 2>/dev/null || true)"
```
[the remark this answers is not in the document]
Comment (samdark) [reply .github/workflows/release.yml]: Addressed in 9b4fd56: the previous-release lookup now also uses `--exclude '*[!0-9.]*'`, so tags containing prerelease suffixes or other non-digit/dot characters are excluded from the changelog boundary.

File: tests/Unit/Packaging/ConfigurationPackagingTest.php [hunk 2]
```diff
@@ -356,6 +356,21 @@ public function documentationWorkflowUsesNightlyBinaryAfterPackageWorkflow(): vo
...
+    public function releaseWorkflowBuildsNotesFromPreviousStableReleaseTag(): void
+    {
+        $workflow = file_get_contents(dirname(__DIR__, 3) . '/.github/workflows/release.yml');
+        self::assertIsString($workflow);
+
+        self::assertStringContainsString("tags:\n      - '*.*.*'", $workflow);
+        self::assertStringContainsString(
+            "git describe --tags --abbrev=0 --match '[0-9]*.[0-9]*.[0-9]*' \"\${tag}^\"",
+            $workflow,
+        );
```
[the remark this answers is not in the document]
Comment (samdark) [reply tests/Unit/Packaging/ConfigurationPackagingTest.php:369]: Addressed in 9b4fd56: the regression test now expects the `--exclude '*[!0-9.]*'` guard in addition to the version-like `--match` pattern.
